src/lib/spriteActor.js

```diff
@@ -0,0 +1,28 @@
+async function loadImage(src) {
+    let img = new Image();
+    img.src = src;
+    return new Promise((res, rej) => {
+        img.onload = () => {
+            res(img);
+        }
+    });
+}
+
+export default class SpriteActor extends Actor{
+    constructor(bounds, properties, states){
```

Comment:
Combine states and properties into a single object

```suggestion
constructor(bounds, spriteObj) {
```